### PR TITLE
crypto ranking app

### PR DESCRIPTION
**Pull request by HeDevedUp**
## Description

**Task:** https://github.com/microverseinc/curriculum-transversal-skills/blob/main/code-review/exercises/full_partner_review.md

This PR makes the following changes:
-  Sets up the GitHub Actions and the [Stylelint](https://stylelint.io/), and [ESLint](https://eslint.org/) linters
-  Sets up the Webpack configuration files and builds scripts

## Type of change 
- [ ]  wrote a unit test for components 
- [X]  used redux query to fetch data from API
- [X] npm install  redux with  @reduxjs/toolkit
- [x] I created a redux directory 
- [X] used filter to filter coin 
- [X] created a component for the currency list
- [X] passed props within a component 
- [X] received props from component
- [X] Added an Input field 
- [X] created a react app with  npx create-react-app my-app
- [X] added linters for react module 
- [X]  added boilerplate for the calculator 
- [X]  used CSS to design the boilerplate 
- [X] refactored the code base and added a component folder
- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [X] My code follows the style guidelines of this project (no linters errors)
- [X] I have checked my code and corrected any misspellings
- [X] I have commented on my code
- [X] I have deleted any unused code as comments in my codebase

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,8 +33,8 @@ To edit this project:
...
 
-## :star2: Show your support
-Give a :star: if you like my project!
+## :star2: Show your support:
+- Give a :star: if you like my project!
 
-## :pencil: License
-This project is [MIT](https://github.com/Gopxfs/fire-data-app/blob/main/LICENSE) licensed.
+## :Acknowledgments:
+- Credit goes to Nelson Sakwa for the design inspiration.
```
Comment (mateo951) [inline README.md:40]: Well done giving appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/) 🤓

File: src/App.js [hunk 2]
```diff
@@ -1,22 +1,25 @@
...
+    <div className="routes">
+      <Router>
+
+        <Routes>
+          <Route path="/" element={<HomePage />} />
+          <Route path="/crypto/:coinId" element={<CrypoDetails />} />
+          <Route path="/profile" element={<Cryptocurrency />} />
+          <Route>404 Not Found</Route>
+        </Routes>
+      </Router>
```
Comment (mateo951) [inline src/App.js]: The project contains a `Home` page and a `Detail` page 🤓 Well done!

File: src/App.js [hunk 3]
```diff
@@ -1,22 +1,25 @@
-import {
-  BrowserRouter as Router, Routes, Route,
-} from 'react-router-dom';
+import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
```
Comment (mateo951) [inline src/App.js:1]: You successfully created a single page application (SPA) using react-router-dom to serve each page without reloading 🙌🔥

File: src/App.css [hunk 4]
```diff
@@ -1,38 +1,522 @@
-.App {
-  text-align: center;
+
```
Comment (mateo951) [inline src/App.css:1]: It's important that your design resembles as much as possible to Nelson's design 🧐 Please be sure to make the necessary changes in your CSS to solve this issue!

Your design 
image

- Things you could improve in your design🤓:
  - Display coins in a grid as shown in the given design (Two coins each row) 
  - Use the same color palette 
  - Follow the general layout of texts and sizes

image

**Review by mateo951 — CHANGES_REQUESTED** [review]
### Hi @HeDevedUp, 

📝👌 **Your project highlights** 😎 
- General Requirements 💪🤓
- App deployed using Netlify🚀
- Implementation of API 🤯
- Home page has a list of categories that can be filtered 🔥
- Great testing 👀 

Good job so far! You still need to work on some issues to prepare your project for the final evaluation, but you are almost there!

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Comment by HeDevedUp** [comment]
thank you for your honest review sir but i chose the first design on the page
[pls check](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)/modules/201496113)

**Comment by mateo951** [comment]
Awesome! So just work on the following: 
- Use the same color palette
- Follow the general layout of texts and sizes

Please be sure that your design is as identical to the design as you can 😎

**Review by algerina — APPROVED** [review]
Hi @HeDevedUp , 
Good job so far! 

May I highlight the following:

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:
- Your project looks awesome:heavy_check_mark:

## STATUS: APPROVED :muscle::clap:

Congratulations! :tada:
Your project is complete!  It's time to merge it :shipit:
Great work on implementing the changes requested by the previous TSEs
 


 ## Optional suggestions
 
 - Please check the comment under the review.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Pull request merged**